RELEASE.md

```diff
@@ -0,0 +1,49 @@
+# Releasing the Conformance Suite
+
+The conformance repository has a release workflow which publishes the test runner
+binary, `connectconformance`, as an artifact that is part of the GitHub release.
+The process was designed to be as friction-free as possible — the result is that
+you just create/tag a release in GitHub, and the creation and upload of artifacts
+is automated after that.
+
+Using the Github UI, [create a new release](https://github.com/connectrpc/conformance/releases/new)
+like so:
+* Under “Choose a tag”, type in “vX.Y.Z” to create a new tag for the release upon publish.
+  Note: The release job does infer the version from the release string and expects versions
+  to start with `v` (e.g. `v0.1.0`).
+* Target the main branch.
+* Title the Release the same as the tag: “vX.Y.Z”.
+* Click “Set as latest release”.
+* If this is a release candidate, or any other kind of pre-release, click "Set as a pre-release".
```

Comment:
Can we specifically mention that pre-1.0 releases are _not_ automatically prereleases? Across our OSS, we settled on treating `v0.9.5` as a regular release. I assume that also applies to RCs for the 1.0, since we _do_ want to feature those releases on the repository homepage?

Reply:
I didn't realize that RCs should not be marked as pre-releases. For Go libraries anyway, using `go get -u` won't get release candidates, so it seemed okay to align the GitHub release state the same way, by marking them as pre-releases.

I can wordsmith a little on this.

Reply:
I added a single statement about pre-v1.0.0 releases. But would like to confirm: do we want release candidates to _not_ be release candidates, too?

Reply:
As far as I can tell, the only effect of this checkbox for Go libraries and binaries is controlling whether or not they're featured on the repository homepage. It seems that we want to advertise the existence of v1.0-rc4, right? We might feel differently about RCs for v2, if we ever get there - we'd want most users to continue using 1.x releases instead of blindly updating to something unstable.